go/vt/vtorc/logic/tablet_discovery.go

```diff
@@ -77,34 +80,51 @@ func updateShardsToWatch() {
 				log.Errorf("Could not parse keyspace/shard %q: %+v", ks, err)
 				continue
 			}
-			newShardsToWatch[k] = append(newShardsToWatch[k], s)
-		} else {
-			ctx, cancel := context.WithTimeout(context.Background(), topo.RemoteOperationTimeout)
-			defer cancel()
-			// Assume this is a keyspace and find all shards in keyspace.
-			// Remove trailing slash if exists.
-			ks = strings.TrimSuffix(ks, "/")
-			shards, err := ts.GetShardNames(ctx, ks)
+			// Parse the shard name into key range value.
+			_, keyRange, err := topo.ValidateShardName(s)
```

Comment:
Your comments seemed to indicate that we don't support arbitrary shard names but only key ranges. This function supports both arbitrary names and key ranges. I think what you might want instead is `key.IsValidKeyRange` and `key.ParseShardingSpec`

Reply:
Done